training/CHAPEL_TUTORIAL.md

```diff
@@ -354,6 +365,44 @@ Expected Output:
 ```console
 0 1 4 9 16 25
 ```
+<a id="coforall"></a>
+## `Coforall` loops
+The second most common parallel loop in arkouda is the [`coforall`](https://chapel-lang.org/docs/users-guide/taskpar/coforall.html) loop.
+The biggest difference between a `coforall` and [`forall`](https://chapel-lang.org/docs/users-guide/datapar/forall.html) loop
+is the way tasks are scheduled.
+
+A `coforall` loop creates one distinct task per loop iteration, each of which executes a copy of the loop body.
+
+But a `forall` loop is a bit more complicated. A forall-loop creates a variable number of tasks to execute the loop
```

Comment:
I think it may be better to explain this in the `forall` section and then refer back to it here. We introduce the `forall` and give examples, but don't really explain it in that section.

Reply:
i feel like getting into how tasks are created in that section will be confusing if we don't have simpler context of `coforall` first. The chapel docs linked to in the forall section go into detail about this (so more experienced readers can explore more, but less experienced readers can wait for the context provided by `coforall`)

Reply:
spoke a bit with ethan about this. We debated switching the order to introduce coforall first, but think it might be putting the wagon before the horse to jump right into talking about task creation. It might be more approachable to talk about an abstract parallel loop first before we can get into the weeds about how things are actually implemented.

We think we're gonna have @TheOddCzar789 take a look and see which he thinks is more intuitive